wellregistry/registry/admin/bulk_upload.py

```diff
@@ -145,17 +151,23 @@ def post(self, request):
             next(data_stream)
             monitoring_locations = []
             error_messages = []
+            warning_messages = []
             row_index = 1
             for row in csv.reader(data_stream):
                 row_index += 1
                 try:
-                    monitoring_locations.append(_get_monitoring_location(row, request.user))
+                    monitoring_locations.append(_get_monitoring_location(row_index,row, request.user, warning_messages))
                 except ValidationError as error:
                     error_messages.append((row_index, error.message_dict))
             if len(error_messages) == 0:
                 MonitoringLocation.objects.bulk_create(monitoring_locations)
-                return redirect(reverse('admin:registry_monitoringlocation_changelist'))
+                if len(warning_messages) == 0:
+                    return redirect(reverse('admin:registry_monitoringlocation_changelist'))
+                else:
+                    warning_messages.insert(0, (0, {'__overall__': 'Data was loaded with the following warnings'}))
+                    
             context['errors'] = error_messages
+            context['warnings'] = warning_messages
```

Comment:
I am not sure why you added warnings as a message separate from errors. Invalid decimals are an error. I don't understand the distinction here.

Reply:
I allows this because the request was to place a null in well_depth when the user had 'null' string in the well_depth field.

Reply:
But the warning message appears if a value is non-null but not a decimal. I think it will be confusing to users to have both an error message and a warning. But that is really for Candice to decide.

Reply:
Ah, I suppose he could remove the warning if there is an error too but in testing he got one or the other.

Reply:
On fields such as latitude or longitude which are required, the code will now generate an error message and a warning message. The well depth field itself will not show this problem because it can be null.